### PR TITLE
[JAX] SBHD reorder skip uses original shape instead of swapped tensor

### PR DESCRIPTION
**Pull request by andrewwhitecdw**
This PR fixes the SBHD Striped skip logic in `tests/jax/test_distributed_fused_attn.py` so it reads the sequence length from the swapped tensor shape rather than the original shape.

## Changes
- `tests/jax/test_distributed_fused_attn.py`: use `tensor.shape[seq_dim]` instead of `shape[seq_dim]` after the SBHD axis swap.

## Details

When `qkv_format == QKVFormat.SBHD`, the test swaps axes 0 and 1 so `seq_dim` becomes 0. The old skip guard read `shape[seq_dim]`, which is the original (unswapped) batch size for SBHD, causing incorrect skips. The fix reads `tensor.shape[seq_dim]`, which reflects the swapped sequence length.

```diff
-            seq_lens = shape[seq_dim]
+            seq_lens = tensor.shape[seq_dim]
```

## Tests
- Covered by the existing parametrized `TestReorderCausalLoadBalancing.test` cases that exercise SBHD inputs.

## Contributor guidelines
- Signed-off-by: Andrew White <andrewwhitecdw@users.noreply.github.com>

### CODE REVIEW

**Comment by cyanguwa** [comment]
@KshitijLakhani, could you please help review this one? Thanks.

**Comment by andrewwhitecdw** [comment]
@greptile-apps thanks for the review. I addressed the test-coverage gap:

- Rewrote `test_sbhd_striped_uses_swapped_seq_dim` to call the parametrized `test(...)` method directly with `QKVFormat.SBHD` and `ReorderStrategy.Striped`, so the skip branch is exercised.
- Monkeypatched `pytest.skip` inside that test so an unexpected skip (which would happen if the original unswapped shape were used) is turned into a failure.

The branch has been amended to a single signed-off commit (ecd70cda) and force-pushed. `python3 -m py_compile tests/jax/test_distributed_fused_attn.py` passed; full pytest was not feasible because JAX is not installed here.

File: tests/jax/test_distributed_fused_attn.py [hunk 1]
```diff
@@ -835,7 +668,7 @@ def test(self, cp_size, shape, qkv_format, reorder_strategy, stripe_size):
             seq_dim = 0
 
         if reorder_strategy == ReorderStrategy.Striped:
-            seq_lens = shape[seq_dim]
+            seq_lens = tensor.shape[seq_dim]
```
Comment (KshitijLakhani) [inline tests/jax/test_distributed_fused_attn.py:838]: Good catch !
This change makes sense to me. I would like to think that the `main` branch basically just skips in the next line due to this incorrect statement (as it would be incorrectly taking the batch value as the seq value) - this explains why our CI never caught this as a failure!
I'd expect those incorrectly skipping tests to not be skipped with this change (and pass)

Comment (KshitijLakhani) [reply tests/jax/test_distributed_fused_attn.py:838]: I did some digging to figure how many and which tests might be skipping and this is the list. A cursory look at TE's CI from release 2.18 does show that these tests were indeed being skipped.

```
 ### L1

  Shape: [3, 32, 8, 64]

  For SBHD, the real sequence length is 32, but the buggy code reads batch size 3.

  Incorrectly skipped:

  - cp_size=4, stripe_size=1
  - cp_size=8, stripe_size=1
  - cp_size=2, stripe_size=4
  - cp_size=4, stripe_size=4
  - cp_size=8, stripe_size=4

  Only cp_size=2, stripe_size=1 currently runs.

  Therefore L1 has five incorrect skips.

  ### L2

  Shape: [4, 32, 12, 32]

  The buggy code reads 4 instead of sequence length 32.

  Incorrectly skipped:

  - cp_size=8, stripe_size=1
  - cp_size=2, stripe_size=4
  - cp_size=4, stripe_size=4
  - cp_size=8, stripe_size=4

  Shape: [1, 16, 1, 1]

  The buggy code reads 1 instead of sequence length 16.

  Incorrectly skipped:

  - cp_size=2, stripe_size=1
  - cp_size=4, stripe_size=1
  - cp_size=8, stripe_size=1
  - cp_size=2, stripe_size=4
  - cp_size=4, stripe_size=4
```

I'd expect these to not be skipped and passed after @andrewwhitecdw 's changes

File: tests/jax/test_distributed_fused_attn.py [hunk 2]
```diff
@@ -848,3 +681,20 @@ def test(self, cp_size, shape, qkv_format, reorder_strategy, stripe_size):
...
+        tensor = tensor.swapaxes(0, 1)  # SBHD: [seq, batch, heads, dim]
+
+        # Old logic read original shape[0]=1 (batch) and skipped; seq_len after swap is 16.
+        reorder = jax.jit(reorder_causal_load_balancing, static_argnums=[1, 2, 3, 4])
+        inverse = jax.jit(inverse_reorder_causal_load_balancing, static_argnums=[1, 2, 3, 4])
+
+        reordered = reorder(tensor, ReorderStrategy.Striped, cp_size, 0, stripe_size)
+        inversed = inverse(reordered, ReorderStrategy.Striped, cp_size, 0, stripe_size)
+
+        assert jnp.array_equal(inversed, tensor)
```
Comment (KshitijLakhani) [inline tests/jax/test_distributed_fused_attn.py]: @andrewwhitecdw I do not think this is needed as the tests above do run SBHD.
With your change to correctly get the seq_lens we should be good.
Please remove this: 
```suggestion
```

**Review by KshitijLakhani — CHANGES_REQUESTED** [review]
Thanks for this PR @andrewwhitecdw ! 
Could you please address the review comments ?
I'm happy to launch our CI on this PR after you've addressed those comments. I'll also confirm that the skipped tests for SBHD actually pass in the CI.

NOTE: A passing CI and approved review are required prior to merge

File: tests/jax/test_distributed_fused_attn.py [hunk 3]
```diff
@@ -848,3 +848,17 @@ def test(self, cp_size, shape, qkv_format, reorder_strategy, stripe_size):
...
+        cp_size = 2
+        shape = (1, 16, 1, 1)  # original [batch, seq, heads, dim]
+
+        # If the skip logic reads the original unswapped batch dim (1), it would skip
+        # because 1 < cp_size * stripe_size. With the fix it reads the swapped seq
+        # dim (16), so the parametrized test should run to completion.
+        monkeypatch.setattr(
+            pytest, "skip", lambda reason: pytest.fail(f"unexpected pytest.skip: {reason}")
+        )
+        self.test(cp_size, shape, QKVFormat.SBHD, ReorderStrategy.Striped, stripe_size)
```
Comment (KshitijLakhani) [inline tests/jax/test_distributed_fused_attn.py]: Re-publishing my comment as it might have become stale as part of an earlier review due to recent commits pushed by @andrewwhitecdw 

I do not think this is needed as the original tests in TestReorderCausalLoadBalancing test() above do run SBHD. With your change to correctly get the seq_lens we should be good.
Please remove this:

```suggestion
```

**Review by KshitijLakhani — CHANGES_REQUESTED** [review]

**Comment by andrewwhitecdw** [comment]
@KshitijLakhani Thanks for the review. I removed the extra regression test as requested — the existing parametrized SBHD Striped cases now exercise the corrected skip logic directly. The branch is a single signed-off commit rebased onto main.

**Comment by KshitijLakhani** [comment]
I launched the L1 tests manually as the `te_ci` pipeline launch from the PR triggered the L0 and L2 tests only. 
Behavior is as expected - no more unintended skips for the SBHD case and they all pass for all intended test levels

**Review by KshitijLakhani — APPROVED** [review]
LGTM !
